Makefile

```diff
@@ -121,10 +121,12 @@ lint: tools
 	CGO_ENABLED=0 tools/bin/revive -formatter friendly -config revive.toml $$($(PACKAGES))
 
 tidy: prepare
+	# tidy isn't a read-only task for go.mod, run FINISH_MOD always, 
+	# so our go.mod1 won't stick in old state
+	trap "$(FINISH_MOD)" EXIT
```

Comment:
this `trap` likely won't work, since every line in the makefile starts a new shell.

Reply:
Oops... So it's a little hard to gracefully run it...